### PR TITLE
Add multi-dim dft test, and fix complex idft

### PR DESCRIPTION
**Pull request by smk2007**
Add multi-dim dft test, and fix complex idft

### CODE REVIEW

File: winml/test/api/LearningModelSessionAPITest.cpp [hunk 1]
```diff
@@ -475,7 +475,135 @@ static void WindowFunction(const wchar_t* window_operator_name, TensorKind kind)
...
+  auto start = std::chrono::high_resolution_clock::now();
+  auto result = session.Evaluate(binding, L"");
+  auto end = std::chrono::high_resolution_clock::now();
+  std::chrono::duration<double, std::micro> evaluate_duration_in_microseconds = end - start;
+  printf("Evaluate Took: %fus\n", evaluate_duration_in_microseconds.count());
+
+  auto error = result.Outputs().Lookup(L"Output.Error").as<TensorFloat>();
+  auto error_ivv = error.GetAsVectorView();
+  for (auto i = 0; i < height * width; i++) {
+    WINML_EXPECT_TRUE(abs(error_ivv.GetAt(i)) < .001);
```
Comment (martinb35) [inline winml/test/api/LearningModelSessionAPITest.cpp:599]: can you give ".001" a descriptively named constant?

File: winml/test/api/LearningModelSessionAPITest.cpp [hunk 2]
```diff
@@ -475,7 +475,135 @@ static void WindowFunction(const wchar_t* window_operator_name, TensorKind kind)
...
+  }
+
+  SaveSoftwareBitmap(L"fft2d.jpg", spectra.SoftwareBitmap());
```
Comment (martinb35) [inline winml/test/api/LearningModelSessionAPITest.cpp:602]: what do you do with these files? I don't see them referenced anywhere else.

File: winml/test/api/LearningModelSessionAPITest.cpp [hunk 3]
```diff
@@ -940,44 +1072,85 @@ static void ModelBuilding_DiscreteFourierTransform() {
...
   // Check results
-  printf("Output.Spectra\n");
-  auto y_tensor = result.Outputs().Lookup(L"Output.Spectra").as<TensorFloat>();
+  auto y_tensor = result.Outputs().Lookup(L"Output.Inverse").as<TensorFloat>();
   auto y_ivv = y_tensor.GetAsVectorView();
-  for (int i = 0; i < output_shape[0] * output_shape[1] * 2; i += 2) {
-    printf("(%f + %fi), ", y_ivv.GetAt(i), y_ivv.GetAt(i + 1));
+  for (uint32_t i = 0; i < y_ivv.Size(); i += 2) {
+    WINML_EXPECT_TRUE(abs(y_ivv.GetAt(i) - input_vector[i / 2]) < .001);
+    WINML_EXPECT_TRUE(abs(y_ivv.GetAt(i + 1) - 0) < .001);
```
Comment (martinb35) [inline winml/test/api/LearningModelSessionAPITest.cpp:1132]: you could re-use that descriptively-named constant here.

File: winml/test/api/LearningModelSessionAPITest.cpp (hunk 2 above)
Comment (smk2007) [reply winml/test/api/LearningModelSessionAPITest.cpp:602]: I look at them to make sure they look correct ;)

File: winml/test/api/LearningModelSessionAPITest.cpp [hunk 4]
```diff
@@ -940,44 +1072,85 @@ static void ModelBuilding_DiscreteFourierTransform() {
...
+  //auto y_tensor = result.Outputs().Lookup(L"Output.Spectra").as<TensorFloat>();
+  //auto y_ivv = y_tensor.GetAsVectorView();
+  //for (uint32_t i = 0; i < y_ivv.Size(); i+=2) {
+  //  auto format_size = 16;
+  //  if (i % format_size == 0 && i != 0) {
+  //    printf("\n");
+  //  }
+  //  printf("(%.2f + %.2fi), ", y_ivv.GetAt(i), y_ivv.GetAt(i + 1));
+  //}
+  //printf("\n");
```
Comment (martinb35) [inline winml/test/api/LearningModelSessionAPITest.cpp:1145]: remove commented out code?

Comment (smk2007) [reply winml/test/api/LearningModelSessionAPITest.cpp:1145]: I use that to debug from time to time, but i can remove and add back when necessary.

File: winml/test/api/LearningModelSessionAPITest.cpp (hunk 3 above)
Comment (smk2007) [reply winml/test/api/LearningModelSessionAPITest.cpp:1132]: constexpr float error_threshold = .001f;

File: winml/test/api/LearningModelSessionAPITest.cpp (hunk 1 above)
Comment (smk2007) [reply winml/test/api/LearningModelSessionAPITest.cpp:599]: constexpr float error_threshold = .001f;

File: winml/test/api/LearningModelSessionAPITest.cpp (hunk 4 above)
Comment (martinb35) [reply winml/test/api/LearningModelSessionAPITest.cpp:1145]: you could leave it in with a comment that says, "this is useful for debugging"

**Review by martinb35 — APPROVED** [review]

**Review by martinb35 — APPROVED** [review]

**Review by martinb35 — APPROVED** [review]

**Pull request merged**